mistletoe/html_renderer.py

```diff
@@ -193,12 +199,20 @@ def render_document(self, token: block_token.Document) -> str:
         inner = '\n'.join([self.render(child) for child in token.children])
         return '{}\n'.format(inner) if inner else ''
 
-    @staticmethod
-    def escape_html(raw: str) -> str:
+    def escape_html_text(self, s: str) -> str:
         """
-        This method is deprecated. Use `html.escape` instead.
+        Like `html.escape()`, but this  looks into the current rendering options
+        to decide which of the quotes (double, single, or both) to escape.
+
+        Intended for escaping text content. To escape content of an attribute,
+        simply call `html.escape()`.
         """
-        return html.escape(raw)
+        s = html.escape(s, quote=False)
+        if self.html_escape_double_quotes:
+            s = s.replace('"', "&quot;")
+        if self.html_escape_single_quotes:
+            s = s.replace('\'', "&#x27;")
```

Comment:
Maybe `&apos;` instead of the ascii value, similar to the double quote?

Reply:
Ah now I see this is how it's done in `html.escape()`. So there's probably a good reason for it. Never mind.

Reply:
Turns out [`&apos;` is an XML entity reference but not HTML](https://fishbowl.pastiche.org/2003/07/01/the_curse_of_apos). Odd.